### PR TITLE
Test/auth signup validation cleanup 

### LINKED ISSUE #7: test: User signup with an existing email
## Purpose of the Test
This test is needed to cover a critical security and data integrity edge case for the user registration process. It ensures that the system prevents duplicate user accounts from being created with the same email address.

## Related Feature/Functionality
-   **Module:** Authentication
-   **Endpoint:** `POST /api/auth/signup`
-   **Controller:** `signup` function in `authController.js`

## Proposed Test Cases
1.  It should reject a signup attempt with an email address that is already in use.

## Current Gaps
The current test suite for authentication (`auth.test.js`) only validates the "happy path" or successful user signup. It does not contain any tests for failure scenarios, such as when a user tries to register with an email that already exists.

## Expected Outcome
The test should confirm that when a `POST` request is sent to `/api/auth/signup` with an existing email, the API correctly returns a `400 Bad Request` status code. It should also verify that no new user is created in the database.

## Additional Notes
-   This test should be added to the `backend/__tests__/auth.test.js` file.
-   The test will need to first create a user in the in-memory database during its setup phase before attempting to register the same user again.

### PR DESCRIPTION
**Pull request by RahulHarihar**
Cleanup: test/auth-signup-validation branch formatting fixes

## Description
This PR cleans up the `test/auth-signup-validation` branch by removing unnecessary formatting noise, including tab-to-space corrections and consistent use of single quotes in imports. No functional changes were made to the code or tests.

## Related Issue
Addresses previous PR #16 and cleanup comment; original issue #7  

## Motivation and Context
The previous PR included formatting changes (tabs, quotes) that made reviewing the actual test logic difficult. This cleanup makes the diff focused only on meaningful changes, improving readability and maintainability.

## Types of Changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Enhancement (improvement to an existing feature)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Test (adds or updates tests only)
- [ ] Documentation (non-code change)

## How Has This Been Tested?
- Ran the existing test suite to ensure all tests still pass.
- Verified that only formatting was changed and no functionality was affected.

## Screenshots (if applicable):
Not applicable (backend test files only).

## Checklist
- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have added tests to cover my changes (existing tests updated for readability)
- [x] All new and existing tests passed

### CODE REVIEW

**Comment by RahulHarihar** [comment]
Hey @archa8, Thanks for your feedback on the previous PR! I’ve cleaned up the branch, so this new PR only includes the essential code and test changes. Appreciate your review!

**Review by archa8 — CHANGES_REQUESTED** [review]
Kindly delete the `package.json`, `package-lock.json` (from the root) and `.env` from the `frontend` folder. Rest is fine

**Review by Avdhesh-Varshney — CHANGES_REQUESTED** [review]
Keep your editor code spacing from 4 to 2-space indentation.
It's very hard to review your changes, and it aligns with the project's formatting standards.

File: backend/__tests__/auth.test.js [hunk 1]
```diff
@@ -7,18 +8,25 @@ const User = require('../models/User');
...
-  server.close();
+	await mongoose.disconnect();
+	await mongoServer.stop();
+	server.close();
 });
 
 describe('Auth API', () => {
+	beforeEach(async () => {
+		await User.deleteMany({});
+	});
```
[the remark this answers is not in the document]
Comment (Avdhesh-Varshney) [reply backend/__tests__/auth.test.js]: image

**Comment by RahulHarihar** [comment]
Hey @archa8, I’ve made the requested changes. Just wanted to check if you’ve had a chance to review them yet.

File: backend/__tests__/auth.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Avdhesh-Varshney) [reply backend/__tests__/auth.test.js]: @RahulHarihar This is still present as it is.
And the indentation is 4-space. Check this once.

**Review by archa8 — APPROVED** [review]
LGTM!

**Comment by RahulHarihar** [comment]
@archa8 thank you, @Avdhesh-Varshney your review?

**Review by Avdhesh-Varshney — APPROVED** [review]
LGTM! Thanks for contribution 🎉 
Keep hacking & keep learning!

**Comment by RahulHarihar** [comment]
@Avdhesh-Varshney Thanks for merging! Appreciate the review
